README.md

```diff
@@ -10,5 +12,99 @@ A tutorial tester script. Extract the code blocks (TODO: also links) from tutori
   * `names` - optional names to give to the extracted snippets, provided as list
   * `extra_roles` - Sphinx or other rst roles which the file might include, so that the parser does not die
   * `parse` - whether to parse the snippet, see below
-  
+
 * `parse_snippet` - parse the snippet into a (prompt, command, result) sequence list. Useful for autoexecuting of docs.
+
+## Getting Started
+
+You can call `tuttest` directly by invoking the `tuttest` command from
+the console. This option might be useful for checking tutorials from
+i.e., Travis CI script. Here is synopsis of a direct `tuttest` call:
+
+```
+  tuttest <file_name> [<snippet_name>]
+```
+  * `<file_name>` is a file in Markdown or RST format, from which you want
+    to extract the code snippets.
+  * `<snippet_name>` is a name that you provided for a snippet. Note that
+    if you do not name a snippet, it will receive the `unnamedX` name, where
+    `X` is the snippet number in the file (starting from 0). Examples of named
+    snippets can be found below.
+
+### Examples
+
+This example presents how to use `tuttest` for extracting named and unnamed
+code snippets from files. Both the Markdown and RST files used in this
+example contain the same code snippets. Therefore the output produced by
+`tuttest` will be the same for both cases.
+
+* `test.rst` (RST format):
+```
+
+.. code-block:: bash
+
+   echo "This is the first unnamed snippet"
+
+
+.. code-block:: bash
+   :name: bash-tutorial
+
+   echo "This is a named snippet"
+   printf "1 + 2 = %d\n" $((1+2))
+
+
+.. code-block:: bash
+
+   echo "This is the second unnamed snippet"
+```
+
+* `test.md` (Markdown format*)
+
+```
+` ``
+echo "This is the first unnamed snippet"
+` ``
+
+<!-- name="bash-tutorial" -->
+` ``
+echo "This is a named snippet"
+printf "1 + 2 = %d\n" $((1+2))
+` ``
+
+` ``
+echo "This is the second unnamed snippet"
+` ``
+```
+  \* Note that you have to change the ``` ` `` ``` used in the script to
+  `` ``` `` before the first script usage.
+
+Here are a ``tuttest`` usage examples. Although the examples shown below
+are based on the RST file, it can be substituted with its Markdown equivalent:
```

Comment:
```suggestion
Here are ``tuttest`` usage examples. For clarity, these examples are run based on the above RST test document:
```

I would avoid repeating once agian the fact that RST and Markdown can both be used.